inc/customizer/controls/react/src/icon-picker/IconPicker.js

```diff
@@ -0,0 +1,70 @@
+import * as icons from '../common/icons';
+import { useState, useEffect } from '@wordpress/element';
+
+const IconPicker = ({ selectIcon, isOpen }) => {
+	const [socialIcons, setSocialIcons] = useState();
+	const [filtered, filterIcons] = useState();
+
+	useEffect(() => {
+		const iconsArr = [];
+		Object.keys(icons).map((icon) => iconsArr.push(icons[icon]));
+		setSocialIcons(iconsArr);
+		filterIcons(iconsArr);
+	}, []);
+
+	useEffect(() => {
+		if (isOpen === true) {
+			const searchInput = document.getElementById('nv--search-input');
+			searchInput.focus();
+		}
+	}, [isOpen]);
+
+	const getIcons = () => {
+		return filtered.map((icon) => {
+			return (
+				// eslint-disable-next-line jsx-a11y/anchor-is-valid
+				<a
+					//
+					href="javascript: return false;"
+					key={icon.id}
+					onClick={() => selectIcon(icon.icon, icon.id)}
+				>
+					{icon.icon}
+				</a>
+			);
+		});
+	};
+
+	const filterHandler = (e) => {
```

Comment:
You should have the `searchQuery` string inside a state variable and filter them inline as I mentioned above.